### ISSUE TITLE
Missing Type and Generator for Mutation Payload

### ISSUE THREAD

**Issue by arnonate** [ardeois/graphql-codegen-typescript-mock-data#42]
Hello! Thanks for the script. I am seeing one issue with my schema. Mutation payloads have an optional `query` field that is not getting generated in the output mocks file. See screenshot below:

![Screen Shot 2021-04-09 at 5 30 11 PM](https://user-images.githubusercontent.com/880025/114246923-88ba9380-9959-11eb-8bd9-ea0cbf496286.png)

![Screen Shot 2021-04-09 at 5 30 30 PM](https://user-images.githubusercontent.com/880025/114246934-8e17de00-9959-11eb-9d5d-db8d6c348f1b.png)

If you can point me in the right direction I will happily contribute a solution. I am just not sure if I have my codegen options set correctly or if this will require a pull request.

```yml
schema:
  - 'schema.grapql':
      headers:
        Authorization: 'Bearer ${AUTH_TOKEN}'
  - 'src/graphql/schema.local.graphql'
documents:
  - ./src/**/*.tsx
  - ./src/**/*.ts
  - ./src/**/*.graphql
overwrite: true
watch: true
generates:
  './src/generated/graphql.tsx':
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
    config:
      skipTypename: false
      withHooks: true
      withHOC: false
      withComponent: false
      withRefetchFn: true
      dedupeOperationSuffix: true
      futureProofEnums: true
      immutableTypes: true
      reactApolloVersion: 3
      # maybeValue: T | null | undefined
  'src/generated/mocks.ts':
    plugins:
      - typescript-mock-data:
          typesFile: './graphql.tsx'
          enumValues: pascal-case#pascalCase
          terminateCircularRelationships: true
          addTypename: true
          typenames: keep
          scalars:
            Date: date
            Datetime: date
```

**Comment by ardeois** [comment]
Hi,
Thanks for the report

I'm not sure to understand your issue, but that's probably a bug on our side.

It's probably related to [this if](https://github.com/ardeois/graphql-codegen-typescript-mock-data/blob/3d2e6791b9ab762540f48037d611b53cf2930320/src/index.ts#L391-L393) that skips the generation of mock data for `Query` and `Mutation` types

**Comment by arnonate** [comment]
Ha! That looks like the culprit for sure! For now I just wrote a small codegen plugin that writes the missing import and type that I need. Is there a reason to skip Query and Mutation types?

**Comment by ardeois** [comment]
Yeah because it didn't make sense to create a mock `aQuery` or `aMutation`
For me, you need to create a mock for a specific query, you don't want to mock the whole `Query` object

So maybe it's the generation of `aQuery()` that don't make sense? not sure

**Comment by naknode** [comment]
> Ha! That looks like the culprit for sure! For now I just wrote a small codegen plugin that writes the missing import and type that I need. Is there a reason to skip Query and Mutation types?

Care to share what you've learned? I, too, am getting this issue and i _understand_ why it's being omitted as what @ardeois stated.

**Comment by ardeois** [comment]
@naknode Can you share your use case as of why you would need `aQuery` ?

**Comment by naknode** [comment]
> @naknode Can you share your use case as of why you would need `aQuery` ?

Sorry, I miswrote. I do _not_ need `aQuery`. As you states, it is not needed for practical purposes. How should I proceed with eliminating a bunch of `Cannot find  name aQuery errors.`

**Comment by naknode** [comment]
> @naknode Can you share your use case as of why you would need `aQuery` ?

@ardeois  Sorry, I miswrote. I do _not_ need `aQuery`. As you states, it is not needed for practical purposes. How should I proceed with eliminating a bunch of `Cannot find  name aQuery errors.`

Great library, btw!

**Comment by ardeois** [comment]
Ah sorry I understand your issue now.
This should be fixed in this plugin. We would probably need to property import `Query` when it's used as a type, but don't generate `aQuery`

Or the easiest way is to keep generating `aQuery` and `aMutation` as it indeed can be used as an output type (as @arnonate shows in his example)

You can do a PR to update this logic https://github.com/ardeois/graphql-codegen-typescript-mock-data/blob/3d2e6791b9ab762540f48037d611b53cf2930320/src/index.ts#L391-L393

**Comment by naknode** [comment]
@ardeois  I am on the latest `1.5.7.` so I should _already_ have the PR you are requesting. I see it in my local `node_modules`, too. So this should already be fixed, no?

Can you show me where @arnonate is using it as a output type? I'm not an expert-level GraphQLer but I understand the fundamentals. :-) Thank you

**Comment by arnonate** [comment]
@naknode I just wrote a small plugin that appends this to the output:

```
module.exports = {
  plugin: (schema, documents, config) => {
    return `// Custom codegen-plugin output
import { Query } from './graphql';
const aQuery = (overrides: Partial<Query>, omit: Set<string>): Query | null | undefined => overrides ? overrides as Query : undefined;`;
  },
};
```

It's not a perfect solution, but I am not actually mocking aQuery anywhere, just needed to get rid of the errors.  Here is the plugin in my `.graphqlrc`:

```
  'src/generated/mocks.ts':
    plugins:
      - typescript-mock-data:
          typesFile: './graphql.tsx'
          enumValues: pascal-case#pascalCase
          terminateCircularRelationships: true
          addTypename: true
          typenames: keep
          scalars:
            Date: date
            Datetime:
              generator: date
              arguments: 'YYYY-MM-DDTHH:mm:ss.SSS\Z'
      - codegen-plugin.js
 ```

**Comment by ardeois** [comment]
@arnonate  fixed in release [1.7.0](https://github.com/ardeois/graphql-codegen-typescript-mock-data/releases/tag/v1.7.0)
